shards/shards.go

```diff
@@ -522,28 +523,38 @@ func (ss *shardedSearcher) StreamSearch(ctx context.Context, q query.Q, opts *zo
 	}
 	defer proc.Release()
 	tr.LazyPrintf("acquired process")
+
+	shards := ss.getShards()
+	if len(shards) == 0 {
+		return nil
+	}
+
 	sender.Send(&zoekt.SearchResult{
 		Stats: zoekt.Stats{
 			Wait: time.Since(start),
 		},
+		Progress: zoekt.Progress{
+			MaxPendingPriority: shards[0].priority,
+		},
 	})
 
-	done, err := ss.streamSearch(ctx, proc, q, opts, stream.SenderFunc(func(event *zoekt.SearchResult) {
+	done, err := ss.streamSearch(ctx, proc, q, opts, shards, stream.SenderFunc(func(event *zoekt.SearchResult) {
 		copyFiles(event)
 		sender.Send(event)
 	}))
 	done()
 	return err
 }
 
-// streamSearch is an internal helper since both Search and StreamSearch are largely similiar.
+// streamSearch is an internal helper since both Search and StreamSearch are
+// largely similar.
 //
 // done must always be called, even if err is non-nil. The SearchResults sent
 // via sender contain references to the underlying mmap data that the garbage
 // collector can't see. Calling done informs the garbage collector it is free
 // to collect those shards. The caller must call copyFiles on any
 // SearchResults it returns/streams out before calling done.
-func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q query.Q, opts *zoekt.SearchOptions, sender zoekt.Sender) (done func(), err error) {
+func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q query.Q, opts *zoekt.SearchOptions, shards []*rankedShard, sender zoekt.Sender) (done func(), err error) {
```

Comment:
we no longer use anything on ss now that we don't need to call ss.getShards. Lets make this a function instead of a method to make that clear.